src/Common/src/Interop/Unix/System.Net.Security.Native/Interop.Gss.cs

```diff
@@ -0,0 +1,336 @@
+            bool isNtlm,
+            SafeGssNameHandle targetName,
+            uint reqFlags,
+            byte[] inputBytes,
+            int inputLength,
+            SafeGssBufferHandle outputToken,
+            out uint retFlags);
+
+        [DllImport(Interop.Libraries.SecurityNative)]
+        internal static extern Status GssAcceptSecContext(
+            out Status minorStatus,
+            ref SafeGssContextHandle contextHandle,
+            SafeGssCredHandle initiatorCredHandle,
+            SafeGssBufferHandle inputToken,
+            SafeGssBufferHandle outputToken,
+            out uint retFlags);
+
+        [DllImport(Interop.Libraries.SecurityNative)]
+        internal static extern Status GssDeleteSecContext(
+            out Status minorStatus,
+            ref IntPtr contextHandle);
+
+        [DllImport(Interop.Libraries.SecurityNative)]
+        internal static extern Status GssWrap(
+            out Status minorStatus,
+            SafeGssContextHandle contextHandle,
+            bool isEncrypt,
+            byte[] inputBytes,
+            int offset,
+            int count,
+            SafeGssBufferHandle outputMessageBuffer);
+
+        [DllImport(Interop.Libraries.SecurityNative)]
+        internal static extern Status GssUnwrap(
+            out Status minorStatus,
+            SafeGssContextHandle contextHandle,
+            byte[] inputBytes,
+            int offset,
+            int count,
+            SafeGssBufferHandle outputMessageBuffer);
+
+        [DllImport(Interop.Libraries.SecurityNative)]
+        internal static extern Status GssInquireSourceName(
+            out Status minorStatus,
+            SafeGssContextHandle contextHandle,
+            out SafeGssNameHandle srcName);
+
+        [DllImport(Interop.Libraries.SecurityNative)]
+        internal static extern Status GssDisplayName(
+            out Status minorStatus,
+            SafeGssNameHandle inputName,
+            SafeGssBufferHandle outputNameBuffer);
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct gss_buffer_desc
+        {
+            internal size_t length;
+            internal IntPtr value;
+        }
+
+        internal enum Status : uint
+        {
+            GSS_S_COMPLETE = 0,
+            GSS_S_CONTINUE_NEEDED = 1
+        }
+
+        [FlagsAttribute]
+        internal enum GssFlags : uint
+        {
+            GSS_C_DELEG_FLAG = 1,
+            GSS_C_MUTUAL_FLAG = 2,
+            GSS_C_REPLAY_FLAG = 4,
+            GSS_C_SEQUENCE_FLAG = 8,
+            GSS_C_CONF_FLAG = 16,
+            GSS_C_INTEG_FLAG = 32,
+            GSS_C_ANON_FLAG = 64,
+            GSS_C_PROT_READY_FLAG = 128,
+            GSS_C_TRANS_FLAG = 256,
+            GSS_C_DCE_STYLE = 4096,
+            GSS_C_IDENTIFY_FLAG = 8192,
+            GSS_C_EXTENDED_ERROR_FLAG = 16384,
+            GSS_C_DELEG_POLICY_FLAG = 32768
+        }
+    }
+}
+
+namespace Microsoft.Win32.SafeHandles
```

Comment:
We thought we needed to be consistent with Interop.Ssl.cs but turns out that is a bad example to follow. @kapilash please move them to a standalone file. @socket, I am assuming location of the file should be same as that of Interop.Gss.cs